### PR TITLE
implement fetching data urls

### PR DESCRIPTION
**Pull request by dylan-conway**
### What does this PR do?

this adds support for fetching data urls
```js
const res = await fetch("data:text/plain;base64,SGVsbG8sIFdvcmxkIQ%3D%3D");
console.log(await res.text()); // Hello, World!
```

### How did you verify your code works?
added tests for fetching valid and invalid data urls

### CODE REVIEW

File: src/bun.js/webcore/response.zig [hunk 1]
```diff
@@ -971,6 +972,45 @@ pub const Fetch = struct {
...
+            .body = Body{
+                .init = Body.Init{
+                    .status_code = 200,
+                },
+                .value = .{
+                    .Blob = blob,
+                },
+            },
+            .allocator = allocator,
+            .status_text = bun.String.create("OK"),
```
Comment (Jarred-Sumner) [inline src/bun.js/webcore/response.zig]: `createAtom`

Comment (Jarred-Sumner) [reply src/bun.js/webcore/response.zig]: or just `.empty`

File: src/resolver/data_url.zig [hunk 2]
```diff
@@ -71,22 +71,28 @@ pub const PercentEncoding = struct {
             }
         }
 
-        if (ret) |some| return allocator.shrink(some, ret_index);
+        if (ret) |some| return allocator.realloc(some, ret_index) catch unreachable;
```
Comment (Jarred-Sumner) [inline src/resolver/data_url.zig]: lets just not reallocate i don't think its necessary

File: src/resolver/data_url.zig [hunk 3]
```diff
@@ -71,22 +71,28 @@ pub const PercentEncoding = struct {
...
     is_base64: bool = false,
 
-    pub fn parse(url: string) ?DataURL {
+    pub fn parse(url: string) !?DataURL {
         if (!strings.startsWith(url, "data:")) {
             return null;
         }
 
-        const comma = strings.indexOfChar(url, ',') orelse return null;
+        var result = try parseWithoutCheck(url);
```
Comment (Jarred-Sumner) [inline src/resolver/data_url.zig:90]: return try parseWithoutCheck(url);

Comment (Jarred-Sumner) [reply src/resolver/data_url.zig:90]: but it doesn't matter

File: src/resolver/data_url.zig [hunk 4]
```diff
@@ -34,7 +34,7 @@ pub const PercentEncoding = struct {
         if (comptime Environment.allow_assert) std.debug.assert(str.len > 0);
         return switch (str[0]) {
             'a'...'z', 'A'...'Z', '0'...'9', '-', '.', '_', '~', '!', '$', '&', '\'', '(', ')', '*', '+', ',', ';', '=', ':', '@' => true,
-            '%' => str.len > 3 and isHex(str[1]) and isHex(str[2]),
+            '%' => str.len >= 3 and isHex(str[1]) and isHex(str[2]),
```
Comment (Jarred-Sumner) [inline src/resolver/data_url.zig:37]: nice

File: src/resolver/data_url.zig (hunk 3 above)
Comment (dylan-conway) [reply src/resolver/data_url.zig:90]: it's actually compiler error, not sure if it's a zig bug

**Pull request merged**